libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx

```diff
@@ -0,0 +1,437 @@
+            .forDisplayForm(getObjRef())
+            .withOptions({
+                ...(searchString ? { filter: searchString } : {}),
+            })
+            .withOffset(offset)
+            .withLimit(limit)
+            .query();
+
+        const mergedValidElements = mergeElementQueryResults(validElements, newElements);
+        const { items } = mergedValidElements;
+
+        // make sure that selected items have both title and uri, otherwise selection in InvertableList won't work
+        // TODO we could maybe use the InvertableList's getItemKey and just use title or uri for example
+        const updatedSelectedItems = updateSelectedOptionsWithData(selectedFilterOptions, items);
+        const updatedPrevSelectedItems = updateSelectedOptionsWithData(prevSelectedFilterOptions, items);
+
+        setSelectedFilterOptions(updatedSelectedItems);
+        setPrevSelectedFilterOptions(updatedPrevSelectedItems);
+        setIsLoading(false);
+        setValidElements(searchString ? newElements : mergedValidElements);
+        setTotalCount(firstLoad ? items.length : totalCount);
+        setFirstLoad(false);
+        setFilterOptions(items as Array<IAttributeElement>);
+    };
+
+    const updateSelectedOptionsWithData = (
+        selection: Array<Partial<IAttributeElement>>,
+        items: AttributeListItem[],
+    ): Array<IAttributeElement> => {
+        const nonEmptyItems = items.filter(isNonEmptyListItem);
+        const createFullItem = (item: Partial<IAttributeElement>): IAttributeElement => ({
+            uri: item.uri ?? "",
+            title: item.title ?? "",
+        });
+
+        return selection.map((selectedItem) => {
+            const foundItem = nonEmptyItems.find(
+                (item) =>
+                    (selectedItem.uri && item.uri === selectedItem.uri) ||
+                    (selectedItem.title && item.title === selectedItem.title),
+            );
+            return foundItem || createFullItem(selectedItem);
+        });
+    };
+
+    /**
+     * getters
+     */
+    const getBackend = () => {
+        return props.backend.withTelemetry("AttributeFilter", props);
+    };
+
+    const getObjRef = (): ObjRef => {
+        const { filter, identifier } = props;
+
+        if (filter && identifier) {
+            throw new Error("Don't use both identifier and filter to specify the attribute to filter");
+        }
+
+        if (filter) {
+            return filterObjRef(filter);
+        }
+
+        if (identifier) {
+            // eslint-disable-next-line no-console
+            console.warn(
+                "Definition of an attribute using 'identifier' is deprecated, use 'filter' property instead. Please see the documentation of [AttributeFilter component](https://sdk.gooddata.com/gooddata-ui/docs/attribute_filter_component.html) for further details.",
+            );
+            return idRef(identifier);
+        }
+    };
+
+    const getElementTotalCount = async (): Promise<void> => {
```

Comment:
This is quite similar to the function with the same name in AttributeDropdown. They could both use an utility function getting `backend`, `workspace`, `objRef` and `searchString` and returnin the Promise<number>